src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py

```diff
@@ -132,6 +132,30 @@ def prepare_pipeline_deployment(
             push_docker_image,
         )
 
+        # if the orchestrator is not running in a local k3d cluster,
```

Comment:
As the kubernetes context is completely arbitrary, it might be 
- pointing to a local k3d cluster
- or even more likely: unspecified but still be pointing to a non-local cluster (even I sometimes run pipelines and just set the kubectl context right before)

I don't really have any better solution to differentiate between local/remote and I think the current implementation is fine, just wanted to mention it

Reply:
As discussed offline, this is going to change in my next PR that implements a new convention for what constitutes "local" Kubeflow orchestrators.